src/Agent.Listener/Configuration/InteractiveSessionConfigurationManager.cs

```diff
@@ -0,0 +1,217 @@
+#if OS_WINDOWS
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(InteractiveSessionConfigurationManager))]
+    public interface IInteractiveSessionConfigurationManager : IAgentService
+    {
+        void Configure(CommandSettings command);
+        void UnConfigure();
+        bool RestartNeeded();        
+    }
+
+    public class InteractiveSessionConfigurationManager : AgentService, IInteractiveSessionConfigurationManager
+    {
+        private ITerminal _terminal;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _terminal = hostContext.GetService<ITerminal>();    
+        }
+
+        public void Configure(CommandSettings command)
+        {
+            var logonAccount = command.GetAutoLogonUserName();
+
+            string domainName;
+            string userName;
+
+            GetAccountSegments(logonAccount, out domainName, out userName);
+
+            if ((string.IsNullOrEmpty(domainName) || domainName.Equals(".", StringComparison.CurrentCultureIgnoreCase)) && !logonAccount.Contains("@"))
+            {
+                logonAccount = String.Format("{0}\\{1}", Environment.MachineName, userName);
+            }
+
+            Trace.Info("LogonAccount after transforming: {0}, user: {1}, domain: {2}", logonAccount, userName, domainName);
+
+            string logonPassword = string.Empty;
+            var windowsServiceHelper = HostContext.GetService<INativeWindowsServiceHelper>();                
+            while (true)
+            {
+                logonPassword = command.GetWindowsLogonPassword(logonAccount);
+                if (windowsServiceHelper.IsValidCredential(domainName, userName, logonPassword))
+                {
+                    Trace.Info("Credential validation succeed");
```

Comment:
refactor dont copy.

---
In reply to: [115659090](https://github.com/Microsoft/vsts-agent/pull/951#discussion_r115659090) [](ancestors = 115659090)